Fish transient prompt hides previous command

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

### What happened?

Since I think 27.0.0 transient prompt stopped working in fish. Before the fix (27.1.0) the prompt didn't work at all, but after it the transient prompt now misses the previous command. By that, I mean that the previous command isn't displayed. Here's the output after running `/usr/bin/ls`:

<img width="157" height="109" alt="Image" src="https://github.com/user-attachments/assets/76c68f8e-2849-43c9-aa98-f6fd5765b21d" />

Additionally, after running `oh-my-posh --version` I get this:

<img width="101" height="55" alt="Image" src="https://github.com/user-attachments/assets/81b0dc29-1f29-4bc9-a637-4f6374a1550a" />

I don't think this is fish's fault because the transient prompt works with starship (but I'm not sure whether they use the built-in transience or not).

### Theme

```
{
    "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
    "transient_prompt": {
        "template": "<#3BD01D>❯</> ",
        "foreground": "#ffffff",
        "background": "transparent"
    },
    "console_title_template": "{{if .Root}}⚡ {{end}}{{ .Shell }} in {{ .Folder }}",
    "blocks": [
        {
            "type": "prompt",
            "alignment": "left",
            "segments": [
                {
                    "properties": {
                        "alpine": "<#0D597F>\uf300</>",
                        "arch": "<#1793d1>\uf303</>",
                        "centos": "\uf304",
                        "debian": "<#A90232>\uf306</>",
                        "elementary": "<#44A2E8>\uf309</>",
                        "fedora": "<#072B61>\uf30a</>",
                        "gentoo": "<#61538D>\uf30d</>",
                        "linux": "<#F5C11B>\ue712</>",
                        "macos": "\ue711",
                        "manjaro": "<#34BE5B>\uf312</>",
                        "mint": "<#86BE43>\uf30f</>",
                        "opensuse": "<73BA25>\uf314</>",
                        "raspbian": "<#C51A4A>\uf315</>",
                        "ubuntu": "<#E95420>\uf31c</>",
                        "windows": "<#26C6DA>\ue70f</>"
                    },
                    "style": "diamond",
                    "leading_diamond": "\ue0b6",
                    "template": " {{ if .WSL }}<#26C6DA>\ue70f</>/{{ end }}{{.Icon}} ",
                    "foreground": "#d2ff5e",
                    "background": "#18354F",
                    "type": "os"
                },
                {
                    "properties": {
                        "home_icon": "󰋜",
                        "mapped_locations": {
                            "~/.config": "\ue615",
                            "~/Desktop": "󰇄",
                            "~/Documents": "󰈙",
                            "~/Documents/CPlusPlus": "\ue61d",
                            "~/Documents/GitHUB": "\uf408",
                            "~/Documents/cpp": "\ue61d",
                            "~/Downloads": "\uf019",
                            "~/Music": "\uf001",
                            "~/Pictures": "󰋫",
                            "~/Videos": "\uf03d"
                        },
                        "style": "full"
                    },
                    "style": "powerline",
                    "foreground": "#38DE21",
                    "powerline_symbol": "\ue0b0",
                    "background": "#2e6f95",
                    "type": "path"
                },
                {
                    "properties": {
                        "branch_icon": "\uf126 ",
                        "fetch_stash_count": true,
                        "fetch_status": true,
                        "fetch_upstream_icon": true
                    },
                    "style": "powerline",
                    "foreground": "#f7fff7",
                    "powerline_symbol": "\ue0b0",
                    "background": "#4ecdc4",
                    "type": "git",
                    "background_templates": [
                        "{{ if or (.Working.Changed) (.Staging.Changed) }}#EDC809{{ end }}",
                        "{{ if and (gt .Ahead 0) (gt .Behind 0) }}#F26D50{{ end }}",
                        "{{ if gt .Ahead 0 }}#B56576{{ end }}",
                        "{{ if gt .Behind 0 }}#6D597A{{ end }}"
                    ]
                },
                {
                    "properties": {
                        "fetch_status": true
                    },
                    "style": "powerline",
                    "foreground": "#193549",
                    "powerline_symbol": "\ue0b0",
                    "background": "#ffeb3b",
                    "type": "svn"
                }
            ],
            "newline": true
        },
        {
            "type": "prompt",
            "alignment": "right",
            "segments": [
                {
                    "style": "powerline",
                    "foreground": "#111111",
                    "powerline_symbol": "\ue0b2",
                    "background": "#ffff66",
                    "type": "root",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "fetch_package_manager": true
                    },
                    "style": "powerline",
                    "template": "{{ if .PackageManagerIcon }}{{ .PackageManagerIcon }} {{else}} 󰎙 {{ end }}{{ .Full }}",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#689f63",
                    "type": "node",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "template": " 🌊 {{ .Full }} ",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#002e53",
                    "type": "nx",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#03589c",
                    "powerline_symbol": "\ue0b2",
                    "background": "#2bb1ee",
                    "type": "dart",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#4063D8",
                    "type": "php",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "template": " R {{ .Full }} ",
                    "foreground": "blue",
                    "powerline_symbol": "\ue0b2",
                    "background": "lightWhite",
                    "type": "r",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#c3002f",
                    "type": "angular"
                },
                {
                    "properties": {
                        "fetch_version": true
                    },
                    "style": "powerline",
                    "foreground": "#111111",
                    "powerline_symbol": "\ue0b2",
                    "background": "#00acd7",
                    "type": "go",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "fetch_version": true
                    },
                    "style": "powerline",
                    "foreground": "#111111",
                    "powerline_symbol": "\ue0b2",
                    "background": "#4063D8",
                    "type": "julia",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#000000",
                    "powerline_symbol": "\ue0b0",
                    "background": "#ebcc34",
                    "type": "kubectl"
                },
                {
                    "properties": {
                        "display_mode": "files",
                        "fetch_version": true,
                        "fetch_virtual_env": false
                    },
                    "style": "powerline",
                    "foreground": "#4584B6",
                    "powerline_symbol": "\ue0b2",
                    "background": "#FFDE57",
                    "type": "python",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#906cff",
                    "type": "kotlin",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#4063D8",
                    "type": "crystal",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "display_mode": "files",
                        "fetch_version": true
                    },
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#AE1401",
                    "type": "ruby",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "fetch_version": true
                    },
                    "style": "powerline",
                    "foreground": "#E64747",
                    "powerline_symbol": "\ue0b2",
                    "background": "#29315A",
                    "type": "java",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "fetch_version": true
                    },
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#4E2BCD",
                    "type": "dotnet",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "template": " \ue7a8 {{ .Full }} ",
                    "foreground": "#193549",
                    "powerline_symbol": "\ue0b2",
                    "background": "#99908a",
                    "type": "rust",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "template": " \ue620 {{ .Full }} ",
                    "foreground": "#fff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#000080",
                    "type": "lua",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "template": " \ue61d cmake {{ .Full }} ",
                    "foreground": "#E8EAEE",
                    "powerline_symbol": "\ue0b2",
                    "background": "#1E9748",
                    "type": "cmake",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "display_mode": "files",
                        "fetch_version": false
                    },
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "#FEAC19",
                    "type": "azfunc",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "display_default": false
                    },
                    "style": "powerline",
                    "foreground": "#ffffff",
                    "powerline_symbol": "\ue0b2",
                    "background": "",
                    "type": "aws",
                    "background_templates": [
                        "{{if contains \"default\" .Profile}}#FFA400{{end}}",
                        "{{if contains \"jan\" .Profile}}#f1184c{{end}}"
                    ],
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#fefefe",
                    "powerline_symbol": "\ue0b2",
                    "background": "#5c41e2",
                    "type": "terraform",
                    "invert_powerline": true
                },
                {
                    "style": "powerline",
                    "foreground": "#FFFFFF",
                    "powerline_symbol": "\ue0b2",
                    "background": "#0077C2",
                    "type": "shell",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "style": "austin"
                    },
                    "style": "powerline",
                    "foreground": "#293241",
                    "powerline_symbol": "\ue0b2",
                    "background": "#ffe66d",
                    "type": "executiontime",
                    "invert_powerline": true
                },
                {
                    "properties": {
                        "always_enabled": true
                    },
                    "style": "diamond",
                    "template": " {{ if gt .Code 0 }}\uf00d{{ else }}\uf42e{{ end }} ",
                    "foreground": "#d3d7cf",
                    "powerline_symbol": "\ue0b0",
                    "background": "#3BD01D",
                    "type": "exit",
                    "background_templates": [
                        "{{ if gt .Code 0 }}#FF005D{{ end }}"
                    ],
                    "invert_powerline": true
                }
            ]
        },
        {
            "type": "prompt",
            "alignment": "left",
            "segments": [
                {
                    "style": "plain",
                    "template": "❯",
                    "foreground": "#3BD01D",
                    "background": "",
                    "type": "text"
                }
            ],
            "newline": true
        }
    ],
    "version": 3,
    "final_space": true
}
```

### What OS are you seeing the problem on?

Linux

### Which shell are you using?

fish

### Log output

```Shell
Version: 27.2.0

Shell: shell (4.1.2)

Prompt:


   󰋜                                                                                                          shell   
❯ 

Segments:

ConsoleTitle(true)                         -   0 ms
Os(true)                                   -   0 ms
Path(true)                                 -   0 ms
Git(false)                                 -   0 ms
Svn(false)                                 -   0 ms
Root(false)                                -   0 ms
Node(false)                                -   0 ms
Nx(false)                                  -   0 ms
Dart(false)                                -   0 ms
Php(false)                                 -   0 ms
R(false)                                   -   0 ms
Angular(false)                             -   0 ms
Go(false)                                  -   0 ms
Julia(false)                               -   0 ms
Kubectl(false)                             -   0 ms
Python(false)                              -   0 ms
Kotlin(false)                              -   0 ms
Crystal(false)                             -   0 ms
Ruby(false)                                -   0 ms
Java(false)                                -   0 ms
Dotnet(false)                              -   0 ms
Rust(false)                                -   0 ms
Lua(false)                                 -   0 ms
Cmake(false)                               -   0 ms
Azfunc(false)                              -   0 ms
Aws(false)                                 -   0 ms
Terraform(false)                           -   0 ms
Shell(true)                                -   0 ms
Executiontime(false)                       -   0 ms
Exit(true)                                 -   0 ms
Text(true)                                 -   0 ms

Run duration: 5.028708ms

Cache path: /home/kamack38/.cache/oh-my-posh

Config path: /home/kamack38/.config/oh-my-posh/kamack.omp.json

Logs:

[DEBUG] 20:52:40.770 debug.go:39 → logging enabled, raw mode: true
[DEBUG] 20:52:40.770 store.go:]:137 → (session) key not found: prompt_count_cache
[TRACE] 20:52:40.770 store.go:](session prompt_count_cache) - 28.303µs
[TRACE] 20:52:40.770 terminal.go:setPromptCount() - 34.304µs
[TRACE] 20:52:40.770 terminal.go:GOOS() - 90ns
[DEBUG] 20:52:40.770 terminal.go:setPwd:102 → /home/kamack38
[TRACE] 20:52:40.770 terminal.go:setPwd() - 15.609µs
[TRACE] 20:52:40.770 terminal.go:Init() - 56.797µs
[TRACE] 20:52:40.770 init.go:SessionID() - 661ns
[TRACE] 20:52:40.770 path.go:Path() - 6.843µs
[DEBUG] 20:52:40.770 store.go:init:94 → (session) loading CONFIG_SOURCE
[DEBUG] 20:52:40.770 store.go:init:94 → (session) loading template_cache
[DEBUG] 20:52:40.770 store.go:init:94 → (session) loading prompt_count_cache
[DEBUG] 20:52:40.770 store.go:init:94 → (session) loading CONFIG
[DEBUG] 20:52:40.770 store.go:init:94 → (session) loading toggle_cache
[TRACE] 20:52:40.770 store.go:init(session fish.c71079a5-e018-4ffc-8437-810794151501.omp.cache) - 155.391µs
[TRACE] 20:52:40.770 path.go:Path() - 71ns
[DEBUG] 20:52:40.770 store.go:init:94 → (device) loading INITVERSIONFISH
[DEBUG] 20:52:40.770 store.go:init:94 → (device) loading DSC_SHELL
[DEBUG] 20:52:40.770 store.go:init:94 → (device) loading DSC_CONFIGURATION
[DEBUG] 20:52:40.770 store.go:init:94 → (device) loading environment_platform
[DEBUG] 20:52:40.770 store.go:init:94 → (device) loading is_wsl
[TRACE] 20:52:40.770 store.go:init(device omp.cache) - 209.342µs
[DEBUG] 20:52:40.770 store.go:]:137 → (device) key not found: reload
[TRACE] 20:52:40.770 store.go:](device reload) - 9.759µs
[DEBUG] 20:52:40.770 store.go:]:150 → (session) found entry: CONFIG - /gH5fwMBAQZDb25maWcB/4AAAR8BB1BhbGV0dGUB/4IAAQtEZWJ1Z1Byb21wdAH/hAABA1ZhcgH/jgABCFBhbGV0dGVzAf+QAAEJVmFsaWRMaW5lAf+EAAEPU2Vjb25kYXJ5UHJvbXB0Af+EAAEPVHJhbnNpZW50UHJvbXB0Af+EAAEJRXJyb3JMaW5lAf+EAAEETWFwcwH/lAABB1VwZ3JhZGUB/5gAAQdFeHRlbmRzAQwAAQtBY2NlbnRDb2xvcgEMAAEUQ29uc29sZVRpdGxlVGVtcGxhdGUBDAABA1BXRAEMAAEGU291cmNlAQwAAQZGb3JtYXQBDAABElRlcm1pbmFsQmFja2dyb3VuZAEMAAEOVG9vbFRpcHNBY3Rpb24BDAABBkJsb2NrcwH/ngABBUN5Y2xlAf+iAAENSVRlcm1GZWF0dXJlcwH/pAABCFRvb2x0aXBzAf+cAAEHVmVyc2lvbgEEAAENTWlncmF0ZUdseXBocwECAAEFQXN5bmMBAgABEFNoZWxsSW50ZWdyYXRpb24BAgABCkZpbmFsU3BhY2UBAgABDVVwZ3JhZGVOb3RpY2UBAgABDlBhdGNoUHdzaEJsZWVkAQIAAQtBdXRvVXBncmFkZQECAAEXRW5hYmxlQ3Vyc29yUG9zaXRpb25pbmcBAgAAABf/gQQBAQdQYWxldHRlAf+CAAEMAQwAAP4CCP+DAwEBB1NlZ21lbnQB/4QAASEBClByb3BlcnRpZXMB/4YAAQVDYWNoZQH/iAABBUFsaWFzAQwAAQ5MZWFkaW5nRGlhbW9uZAEMAAEPVHJhaWxpbmdEaWFtb25kAQwAAQhUZW1wbGF0ZQEMAAEKRm9yZWdyb3VuZAEMAAEOVGVtcGxhdGVzTG9naWMBDAABD1Bvd2VybGluZVN5bWJvbAEMAAEKQmFja2dyb3VuZAEMAAEGRmlsbGVyAQwAAQRUeXBlAQwAAQVTdHlsZQEMAAEWTGVhZGluZ1Bvd2VybGluZVN5bWJvbAEMAAETRm9yZWdyb3VuZFRlbXBsYXRlcwH/igABBFRpcHMB/4wAARNCYWNrZ3JvdW5kVGVtcGxhdGVzAf+KAAEJVGVtcGxhdGVzAf+KAAEORXhjbHVkZUZvbGRlcnMB/4wAAQ5JbmNsdWRlRm9sZGVycwH/jAABBU5lZWRzAf+MAAEHVGltZW91dAEEAAEITWF4V2lkdGgBBAABCE1pbldpZHRoAQQAAQhEdXJhdGlvbgEEAAEKTmFtZUxlbmd0aAEEAAEFSW5kZXgBBAABC0ludGVyYWN0aXZlAQIAAQdFbmFibGVkAQIAAQdOZXdsaW5lAQIAAQ9JbnZlcnRQb3dlcmxpbmUBAgABBUZvcmNlAQIAAQdUb2dnbGVkAQIAAAAT/4UEAQEDTWFwAf+GAAEMARAAAC3/hwMBAQVDYWNoZQH/iAABAgEIRHVyYXRpb24BDAABCFN0cmF0ZWd5AQwAAAAS/4kCAQEETGlzdAH/igABDAAAFv+LAgEBCFtdc3RyaW5nAf+MAAEMAAAn/40EAQEXbWFwW3N0cmluZ11pbnRlcmZhY2Uge30B/44AAQwBEAAALf+PAwEBCFBhbGV0dGVzAf+QAAECAQRMaXN0Af+SAAEIVGVtcGxhdGUBDAAAACn/kQQBARhtYXBbc3RyaW5nXWNvbG9yLlBhbGV0dGUB/5IAAQwB/4IAAD//kwMBAQZDb25maWcB/5QAAQMBCFVzZXJOYW1lAf+WAAEISG9zdE5hbWUB/5YAAQlTaGVsbE5hbWUB/5YAAAAT/5UEAQEDTWFwAf+WAAEMAQwAAFz/lwMBAQZDb25maWcB/5gAAQYBBlNvdXJjZQEMAAEISW50ZXJ2YWwBDAABBkxhdGVzdAEMAAEEQXV0bwECAAENRGlzcGxheU5vdGljZQECAAEFRm9yY2UBAgAAAB7/nQIBAQ9bXSpjb25maWcuQmxvY2sB/54AAf+aAAD/qv+ZAwEC/5oAAQwBBFR5cGUBDAABCUFsaWdubWVudAEMAAEGRmlsbGVyAQwAAQhPdmVyZmxvdwEMAAEOTGVhZGluZ0RpYW1vbmQBDAABD1RyYWlsaW5nRGlhbW9uZAEMAAEIU2VnbWVudHMB/5wAAQhNYXhXaWR0aAEEAAEITWluV2lkdGgBBAABB05ld2xpbmUBAgABBUZvcmNlAQIAAQVJbmRleAEEAAAAIP+bAgEBEVtdKmNvbmZpZy5TZWdtZW50Af+cAAH/hAAAFP+hAgEBBUN5Y2xlAf+iAAH/oAAAKv+fAwEC/6AAAQIBCkJhY2tncm91bmQBDAABCkZvcmVncm91bmQBDAAAABv/owIBAQ1JVGVybUZlYXR1cmVzAf+kAAEMAAD+Dan/gAcGEDwjM0JEMDFEPuKdrzwvPiABByNmZmZmZmYDC3RyYW5zcGFyZW50AAMBA2NkbgEEMTY4aAADNHt7aWYgLlJvb3R9feKaoSB7e2VuZH19e3sgLlNoZWxsIH19IGluIHt7IC5Gb2xkZXIgfX0CMS9ob21lL2thbWFjazM4Ly5jb25maWcvb2gtbXktcG9zaC9rYW1hY2sub21wLmpzb24BBGpzb24DAwEGcHJvbXB0AQRsZWZ0BQQBDwRtaW50BnN0cmluZwwRAA88Izg2QkU0Mz7vjI88Lz4Ib3BlbnN1c2UGc3RyaW5nDBAADjw3M0JBMjU+74yUPC8+BW1hY29zBnN0cmluZwwFAAPunJEHd2luZG93cwZzdHJpbmcMEQAPPCMyNkM2REE+7pyPPC8+BmFscGluZQZzdHJpbmcMEQAPPCMwRDU5N0Y+74yAPC8+BmNlbnRvcwZzdHJpbmcMBQAD74yECmVsZW1lbnRhcnkGc3RyaW5nDBEADzwjNDRBMkU4Pu+MiTwvPgZmZWRvcmEGc3RyaW5nDBEADzwjMDcyQjYxPu+MijwvPgZnZW50b28Gc3RyaW5nDBEADzwjNjE1MzhEPu+MjTwvPgVsaW51eAZzdHJpbmcMEQAPPCNGNUMxMUI+7pySPC8+CHJhc3BiaWFuBnN0cmluZwwRAA88I0M1MUE0QT7vjJU8Lz4HbWFuamFybwZzdHJpbmcMEQAPPCMzNEJFNUI+74ySPC8+BnVidW50dQZzdHJpbmcMEQAPPCNFOTU0MjA+74ycPC8+BGFyY2gGc3RyaW5nDBEADzwjMTc5M2QxPu+MgzwvPgZkZWJpYW4Gc3RyaW5nDBEADzwjQTkwMjMyPu+MhjwvPgMD7oK2AjEge3sgaWYgLldTTCB9fTwjMjZDNkRBPu6cjzwvPi97eyBlbmQgfX17ey5JY29ufX0gAQcjZDJmZjVlAwcjMTgzNTRGAgJvcwEHZGlhbW9uZAABAwlob21lX2ljb24Gc3RyaW5nDAYABPOwi5wQbWFwcGVkX2xvY2F0aW9ucxdtYXBbc3RyaW5nXWludGVyZmFjZSB7ff+O/gESAAoIfi9WaWRlb3MGc3RyaW5nDAUAA++AvQl+Ly5jb25maWcGc3RyaW5nDAUAA+6YlQl+L0Rlc2t0b3AGc3RyaW5nDAYABPOwh4QSfi9Eb2N1bWVudHMvR2l0SFVCBnN0cmluZwwFAAPvkIgKfi9QaWN0dXJlcwZzdHJpbmcMBgAE87CLqwt+L0RvY3VtZW50cwZzdHJpbmcMBgAE87CImRV+L0RvY3VtZW50cy9DUGx1c1BsdXMGc3RyaW5nDAUAA+6YnQ9+L0RvY3VtZW50cy9jcHAGc3RyaW5nDAUAA+6YnQt+L0Rvd25sb2FkcwZzdHJpbmcMBQAD74CZB34vTXVzaWMGc3RyaW5nDAUAA++AgQVzdHlsZQZzdHJpbmcMBgAEZnVsbAYHIzM4REUyMQID7oKwAQcjMmU2Zjk1AgRwYXRoAQlwb3dlcmxpbmUAAQQLYnJhbmNoX2ljb24Gc3RyaW5nDAYABO+EpiARZmV0Y2hfc3Rhc2hfY291bnQEYm9vbAICAAEMZmV0Y2hfc3RhdHVzBGJvb2wCAgABE2ZldGNoX3Vwc3RyZWFtX2ljb24EYm9vbAICAAEGByNmN2ZmZjcCA+6CsAEHIzRlY2RjNAIDZ2l0AQlwb3dlcmxpbmUEBEF7eyBpZiBvciAoLldvcmtpbmcuQ2hhbmdlZCkgKC5TdGFnaW5nLkNoYW5nZWQpIH19I0VEQzgwOXt7IGVuZCB9fTl7eyBpZiBhbmQgKGd0IC5BaGVhZCAwKSAoZ3QgLkJlaGluZCAwKSB9fSNGMjZENTB7eyBlbmQgfX0ke3sgaWYgZ3QgLkFoZWFkIDAgfX0jQjU2NTc2e3sgZW5kIH19JXt7IGlmIGd0IC5CZWhpbmQgMCB9fSM2RDU5N0F7eyBlbmQgfX0AAQEMZmV0Y2hfc3RhdHVzBGJvb2wCAgABBgcjMTkzNTQ5AgPugrABByNmZmViM2ICA3N2bgEJcG93ZXJsaW5lAAMBAAEGcHJvbXB0AQVyaWdodAUZBwcjMTExMTExAgPugrIBByNmZmZmNjYCBHJvb3QBCXBvd2VybGluZRIBAAEBFWZldGNoX3BhY2thZ2VfbWFuYWdlcgRib29sAgIAAQVYe3sgaWYgLlBhY2thZ2VNYW5hZ2VySWNvbiB9fXt7IC5QYWNrYWdlTWFuYWdlckljb24gfX0ge3tlbHNlfX0g87COmSB7eyBlbmQgfX17eyAuRnVsbCB9fQEHI2ZmZmZmZgID7oKyAQcjNjg5ZjYzAgRub2RlAQlwb3dlcmxpbmUSAQAGEiDwn4yKIHt7IC5GdWxsIH19IAEHI2ZmZmZmZgID7oKyAQcjMDAyZTUzAgJueAEJcG93ZXJsaW5lEgEABwcjMDM1ODljAgPugrIBByMyYmIxZWUCBGRhcnQBCXBvd2VybGluZRIBAAcHI2ZmZmZmZgID7oKyAQcjNDA2M0Q4AgNwaHABCXBvd2VybGluZRIBAAYPIFIge3sgLkZ1bGwgfX0gAQRibHVlAgPugrIBCmxpZ2h0V2hpdGUCAXIBCXBvd2VybGluZRIBAAcHI2ZmZmZmZgID7oKyAQcjYzMwMDJmAgdhbmd1bGFyAQlwb3dlcmxpbmUAAQENZmV0Y2hfdmVyc2lvbgRib29sAgIAAQYHIzExMTExMQID7oKyAQcjMDBhY2Q3AgJnbwEJcG93ZXJsaW5lEgEAAQENZmV0Y2hfdmVyc2lvbgRib29sAgIAAQYHIzExMTExMQID7oKyAQcjNDA2M0Q4AgVqdWxpYQEJcG93ZXJsaW5lEgEABwcjMDAwMDAwAgPugrABByNlYmNjMzQCB2t1YmVjdGwBCXBvd2VybGluZQABAwxkaXNwbGF5X21vZGUGc3RyaW5nDAcABWZpbGVzDWZldGNoX3ZlcnNpb24EYm9vbAICAAERZmV0Y2hfdmlydHVhbF9lbnYEYm9vbAICAAAGByM0NTg0QjYCA+6CsgEHI0ZGREU1NwIGcHl0aG9uAQlwb3dlcmxpbmUSAQAHByNmZmZmZmYCA+6CsgEHIzkwNmNmZgIGa290bGluAQlwb3dlcmxpbmUSAQAHByNmZmZmZmYCA+6CsgEHIzQwNjNEOAIHY3J5c3RhbAEJcG93ZXJsaW5lEgEAAQIMZGlzcGxheV9tb2RlBnN0cmluZwwHAAVmaWxlcw1mZXRjaF92ZXJzaW9uBGJvb2wCAgABBgcjZmZmZmZmAgPugrIBByNBRTE0MDECBHJ1YnkBCXBvd2VybGluZRIBAAEBDWZldGNoX3ZlcnNpb24EYm9vbAICAAEGByNFNjQ3NDcCA+6CsgEHIzI5MzE1QQIEamF2YQEJcG93ZXJsaW5lEgEAAQENZmV0Y2hfdmVyc2lvbgRib29sAgIAAQYHI2ZmZmZmZgID7oKyAQcjNEUyQkNEAgZkb3RuZXQBCXBvd2VybGluZRIBAAYRIO6eqCB7eyAuRnVsbCB9fSABByMxOTM1NDkCA+6CsgEHIzk5OTA4YQIEcnVzdAEJcG93ZXJsaW5lEgEABhEg7pigIHt7IC5GdWxsIH19IAEEI2ZmZgID7oKyAQcjMDAwMDgwAgNsdWEBCXBvd2VybGluZRIBAAYXIO6YnSBjbWFrZSB7eyAuRnVsbCB9fSABByNFOEVBRUUCA+6CsgEHIzFFOTc0OAIFY21ha2UBCXBvd2VybGluZRIBAAECDGRpc3BsYXlfbW9kZQZzdHJpbmcMBwAFZmlsZXMNZmV0Y2hfdmVyc2lvbgRib29sAgIAAAYHI2ZmZmZmZgID7oKyAQcjRkVBQzE5AgZhemZ1bmMBCXBvd2VybGluZRIBAAEBD2Rpc3BsYXlfZGVmYXVsdARib29sAgIAAAYHI2ZmZmZmZgID7oKyAwNhd3MBCXBvd2VybGluZQQCMHt7aWYgY29udGFpbnMgImRlZmF1bHQiIC5Qcm9maWxlfX0jRkZBNDAwe3tlbmR9fSx7e2lmIGNvbnRhaW5zICJqYW4iIC5Qcm9maWxlfX0jZjExODRje3tlbmR9fQ4BAAcHI2ZlZmVmZQID7oKyAQcjNWM0MWUyAgl0ZXJyYWZvcm0BCXBvd2VybGluZRIBAAcHI0ZGRkZGRgID7oKyAQcjMDA3N0MyAgVzaGVsbAEJcG93ZXJsaW5lEgEAAQEFc3R5bGUGc3RyaW5nDAgABmF1c3RpbgYHIzI5MzI0MQID7oKyAQcjZmZlNjZkAg1leGVjdXRpb250aW1lAQlwb3dlcmxpbmUSAQABAQ5hbHdheXNfZW5hYmxlZARib29sAgIAAQUuIHt7IGlmIGd0IC5Db2RlIDAgfX3vgI17eyBlbHNlIH1975Cue3sgZW5kIH19IAEHI2QzZDdjZgID7oKwAQcjM0JEMDFEAgRleGl0AQdkaWFtb25kBAEje3sgaWYgZ3QgLkNvZGUgMCB9fSNGRjAwNUR7eyBlbmQgfX0OAQAAAQZwcm9tcHQBBGxlZnQFAQYD4p2vAQcjM0JEMDFEBQR0ZXh0AQVwbGFpbgADAQAEBgQBAA==
[TRACE] 20:52:40.770 store.go:](session CONFIG) - 33.333µs
[TRACE] 20:52:40.770 gob.go:Restore() - 160.971µs
[TRACE] 20:52:40.770 gob.go:Get() - 202.509µs
[TRACE] 20:52:40.770 terminal.go:Shell() - 80ns
[TRACE] 20:52:40.770 terminal.go:Flags() - 50ns
[DEBUG] 20:52:40.770 store.go:]:150 → (session) found entry: template_cache - {map[Exit:0xc00030e640 Os:0xc00036d2c0 Path:0xc000232180 Shell:0xc00030e730 Text:0xc0003770e0] map[] ~ kamack38  kamack38 rogstrixscar-g15 4.1.2 fish /home/kamack38 garuda 27.2.0 12 1 0 0 false false}
[TRACE] 20:52:40.771 store.go:](session template_cache) - 35.717µs
[TRACE] 20:52:40.771 cache.go:restoreCache() - 54.742µs
[DEBUG] 20:52:40.771 writer.go:Init:107 → terminal program: ghostty
[DEBUG] 20:52:40.771 writer.go:Init:108 → terminal shell: shell
[DEBUG] 20:52:40.771 terminal.go:Getenv:64 → NO DATA
[TRACE] 20:52:40.771 terminal.go:Getenv(OMP_CACHE_DISABLED) - 3.426µs
[DEBUG] 20:52:40.771 store.go:]:137 → (device) key not found: accent_color
[TRACE] 20:52:40.771 store.go:](device accent_color) - 6.262µs
[TRACE] 20:52:40.771 colors.go:SetAccentColor() - 10.018µs
[TRACE] 20:52:40.771 terminal.go:Shell() - 50ns
[DEBUG] 20:52:40.771 terminal.go:Getenv:64 → 4.1.2
[TRACE] 20:52:40.771 terminal.go:Getenv(POSH_SHELL_VERSION) - 3.176µs
[DEBUG] 20:52:40.771 debug.go:PrintDebug:24 → segment: Title
[DEBUG] 20:52:40.771 text.go:patchTemplate:149 → {{if .Root}}⚡ {{end}}{{ .Shell }} in {{ .Folder }}
[TRACE] 20:52:40.771 terminal.go:Flags() - 40ns
[DEBUG] 20:52:40.771 terminal.go:Getenv:64 → NO DATA
[TRACE] 20:52:40.771 terminal.go:Getenv(POSH_CURSOR_LINE) - 2.554µs
[DEBUG] 20:52:40.771 terminal.go:Getenv:64 → NO DATA
[TRACE] 20:52:40.771 terminal.go:Getenv(POSH_CURSOR_COLUMN) - 2.034µs
[TRACE] 20:52:40.771 terminal.go:Flags() - 30ns
[TRACE] 20:52:40.771 terminal.go:Flags() - 40ns
[TRACE] 20:52:40.771 terminal.go:Flags() - 61ns
[TRACE] 20:52:40.771 terminal.go:Flags() - 180ns
[TRACE] 20:52:40.771 terminal.go:Flags() - 270ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 terminal.go:GOOS() - 50ns
[DEBUG] 20:52:40.771 segment.go:Execute:109 → segment: Svn
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.771 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.771 terminal.go:GOOS() - 60ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 60ns
[TRACE] 20:52:40.771 store.go:](session toggle_cache) - 5.751µs
[DEBUG] 20:52:40.771 segment.go:Execute:109 → segment: Git
[DEBUG] 20:52:40.771 segment.go:Execute:109 → segment: Path
[DEBUG] 20:52:40.771 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.771 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.771 store.go:](session toggle_cache) - 7.003µs
[DEBUG] 20:52:40.771 store.go:]:150 → (device) found entry: is_wsl - false
[DEBUG] 20:52:40.771 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.771 store.go:](device is_wsl) - 5.09µs
[DEBUG] 20:52:40.771 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.771 terminal_unix.go:IsWsl() - 13.465µs
[TRACE] 20:52:40.771 terminal_unix.go:IsWsl2() - 16.771µs
[DEBUG] 20:52:40.771 properties.go:GetBool:40 → disable_with_jj: false
[DEBUG] 20:52:40.771 properties.go:GetBool:40 → display_cygpath: false
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 separator.go:Separator() - 60ns
[DEBUG] 20:52:40.771 terminal.go:ResolveSymlink:195 → /home/kamack38
[TRACE] 20:52:40.771 terminal.go:Shell() - 81ns
[TRACE] 20:52:40.771 terminal.go:ResolveSymlink(/home/kamack38) - 16.611µs
[ERROR] 20:52:40.771 terminal.go:HasParentFilePath:464 → stat .git: no such file or directory
[TRACE] 20:52:40.771 terminal.go:HasParentFilePath(.git) - 37.511µs
[TRACE] 20:52:40.771 terminal.go:GOOS() - 41ns
[DEBUG] 20:52:40.771 properties.go:GetBool:40 → mapped_locations_enabled: true
[DEBUG] 20:52:40.771 properties.go:GetString:46 → windows_registry_icon: 
[DEBUG] 20:52:40.771 properties.go:GetString:46 → windows_registry_icon: 
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 60ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[DEBUG] 20:52:40.771 properties.go:GetString:46 → home_icon: 󰋜
[DEBUG] 20:52:40.771 properties.go:GetKeyValueMap:64 → mapped_locations: map[~/.config: ~/Desktop:󰇄 ~/Documents:󰈙 ~/Documents/CPlusPlus: ~/Documents/GitHUB: ~/Documents/cpp: ~/Downloads: ~/Music: ~/Pictures:󰋫 ~/Videos:]
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 31ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 31ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 50ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 60ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 90ns
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[TRACE] 20:52:40.771 separator.go:Separator() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 41ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 separator.go:Separator() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 50ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 separator.go:Separator() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.771 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.771 properties.go:GetStringArray:70 → cycle: []
[TRACE] 20:52:40.771 terminal.go:Flags() - 250ns
[DEBUG] 20:52:40.771 properties.go:GetString:46 → folder_separator_icon: /
[DEBUG] 20:52:40.771 properties.go:GetBool:40 → cycle_folder_separator: false
[DEBUG] 20:52:40.771 properties.go:GetString:46 → folder_format: %s
[DEBUG] 20:52:40.771 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.771 properties.go:GetString:46 → edge_format: %s
[TRACE] 20:52:40.771 terminal.go:GOOS() - 50ns
[DEBUG] 20:52:40.771 properties.go:GetString:46 → left_format: %s
[DEBUG] 20:52:40.771 segment.go:Execute:109 → segment: Os
[DEBUG] 20:52:40.771 properties.go:GetString:46 → right_format: %s
[DEBUG] 20:52:40.771 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.771 terminal.go:Flags() - 40ns
[TRACE] 20:52:40.771 store.go:](session toggle_cache) - 6.493µs
[TRACE] 20:52:40.771 terminal.go:GOOS() - 30ns
[ERROR] 20:52:40.771 terminal.go:CommandPath:308 → exec: "svn": executable file not found in $PATH
[DEBUG] 20:52:40.771 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.771 terminal.go:StackCount() - 60ns
[TRACE] 20:52:40.771 terminal.go:CommandPath(svn) - 75.792µs
[TRACE] 20:52:40.771 terminal.go:GOOS() - 41ns
[TRACE] 20:52:40.771 terminal.go:HasCommand(svn) - 79.789µs
[DEBUG] 20:52:40.771 store.go:]:150 → (device) found entry: environment_platform - garuda
[TRACE] 20:52:40.771 terminal_unix.go:DirIsWritable(/home/kamack38) - 7.444µs
[TRACE] 20:52:40.771 store.go:](device environment_platform) - 6.933µs
[DEBUG] 20:52:40.771 properties.go:GetBool:40 → display_distro_name: false
[DEBUG] 20:52:40.771 properties.go:GetString:46 → garuda: 
[DEBUG] 20:52:40.771 properties.go:GetString:46 → linux: <#F5C11B></>
[DEBUG] 20:52:40.771 text.go:patchTemplate:149 →  {{ if .WSL }}<#26C6DA></>/{{ end }}{{.Data.Icon}} 
[DEBUG] 20:52:40.773 text.go:patchTemplate:149 →  {{ .Data.Path }} 
[TRACE] 20:52:40.773 terminal.go:Shell() - 191ns
[TRACE] 20:52:40.773 terminal.go:Flags() - 260ns
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:Flags() - 150ns
[TRACE] 20:52:40.773 terminal.go:GOOS() - 70ns
[TRACE] 20:52:40.773 terminal.go:Flags() - 330ns
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Node
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:Flags() - 441ns
[TRACE] 20:52:40.773 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 5.871µs
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Dart
[TRACE] 20:52:40.773 terminal.go:GOOS() - 70ns
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:GOOS() - 50ns
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 5.751µs
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.js *.ts package.json .nvmrc pnpm-workspace.yaml .pnpmfile.cjs .vue]
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Root
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.dart pubspec.yaml pubspec.yml pubspec.lock]
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 6.542µs
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: [.dart_tool]
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 9.247µs
[DEBUG] 20:52:40.773 segment.go::Floggled:230 → no toggles found
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.773 terminal.go:Flags() - 40ns
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Php
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal_unix.go:Root() - 692ns
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [workspace.json nx.json]
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: R
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.258µs
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:GOOS() - 31ns
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.php composer.json composer.lock .php-version blade.php]
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.448µs
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.R *.Rmd *.Rsx *.Rda *.Rd *.Rproj .Rproj.user]
[TRACE] 20:52:40.773 terminal.go:Flags() - 70ns
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[TRACE] 20:52:40.773 terminal.go:GOOS() - 40ns
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Julia
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Go
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[TRACE] 20:52:40.773 terminal.go:GOOS() - 40ns
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 3.758µs
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:Flags() - 40ns
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.288µs
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 6.573µs
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → parse_kubeconfig: true
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.jl]
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Python
[DEBUG] 20:52:40.773 terminal.go:Getenv:64 → NO DATA
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:Getenv(KUBECONFIG) - 3.226µs
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 3.787µs
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.go go.mod go.sum go.work go.work.sum]
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 properties.go:GetString:46 → display_mode: files
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[ERROR] 20:52:40.773 terminal.go:FileContent:207 → open /home/kamack38/.kube/config: no such file or directory
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.py *.ipynb pyproject.toml venv.bak]
[TRACE] 20:52:40.773 terminal.go:FileContent(/home/kamack38/.kube/config) - 10.63µs
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Kotlin
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:Flags() - 50ns
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.177µs
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.kt *.kts *.ktm]
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:GOOS() - 40ns
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Ruby
al
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 3.888µs
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 5.952µs
[TRACE] 20:52:40.773 terminal.go:GOOS() - 40ns
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Java
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.rb Rakefile Gemfile]
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.248µs
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.cr shard.yml]
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 terminal.go:Flags() - 40ns
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → display_error: false
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:Flags() - 40ns
[TRACE] 20:52:40.773 terminal.go:GOOS() - 40ns
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Dotnet
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [pom.xml build.gradle.kts build.sbt .java-version .deps.edn project.clj build.boot *.java *.class *.gradle *.jar *.clj *.cljc]
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Rust
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 13.144µs
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[TRACE] 20:52:40.773 terminal.go:Flags() - 60ns
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.208µs
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 terminal.go:Flags() - 40ns
[TRACE] 20:52:40.773 terminal.go:GOOS() - 50ns
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.rs Cargo.toml Cargo.lock]
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Lua
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Cmake
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.058µs
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 6.752µs
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.cs *.csx *.vb *.sln *.slnx *.slnf *.csproj *.vbproj *.fs *.fsx *.fsproj global.json]
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Azfunc
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.cmake CMakeLists.txt]
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 properties.go:GetString:46 → preferred_executable: lua
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 3.627µs
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [*.lua *.rockspec]
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:Flags() - 30ns
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → extensions: [host.json local.settings.json function.json]
[TRACE] 20:52:40.773 terminal.go:GOOS() - 40ns
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: [lua]
[DEBUG] 20:52:40.773 properties.go:GetStringArray:70 → folders: []
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[TRACE] 20:52:40.773 terminal.go:GOOS() - 50ns
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → home_enabled: false
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Terraform
[TRACE] 20:52:40.773 terminal.go:Flags() - 40ns
[DEBUG] 20:52:40.773 store.go:](session toggle_cache) - 4.007µs
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[TRACE] 20:52:40.773 terminal.go:Flags() - 81ns
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 4.187µs
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → display_default: false
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Shell
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[DEBUG] 20:52:40.773 terminal.go:Getenv:64 → NO DATA
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetString:46 → command: terraform
[TRACE] 20:52:40.773 terminal.go:Getenv(AWS_VAULT) - 3.105µs
[TRACE] 20:52:40.773 terminal.go:GOOS() - 60ns
[DEBUG] 20:52:40.773 terminal.go:Getenv:64 → NO DATA
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 terminal.go:Getenv(AWS_DEFAULT_PROFILE) - 7.965µs
[DEBUG] 20:52:40.773 terminal.go:Getenv:64 → NO DATA
[TRACE] 20:52:40.773 terminal.go:Getenv(AWS_PROFILE) - 1.914µs
[DEBUG] 20:52:40.773 properties.go:GetKeyValueMap:64 → mapped_shell_names: map[]
[DEBUG] 20:52:40.773 terminal.go:Getenv:64 → NO DATA
[TRACE] 20:52:40.773 terminal.go:Getenv(AWS_REGION) - 2.274µs
[TRACE] 20:52:40.773 terminal.go:Shell() - 60ns
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[DEBUG] 20:52:40.773 terminal.go:Getenv:64 → NO DATA
[TRACE] 20:52:40.773 terminal.go:Getenv(AWS_DEFAULT_REGION) - 2.595µs
[TRACE] 20:52:40.773 terminal.go:Flags() - 70ns
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 6.903µs
[DEBUG] 20:52:40.773 terminal.go:Getenv:64 → NO DATA
[TRACE] 20:52:40.773 terminal.go:Getenv(AWS_CONFIG_FILE) - 3.145µs
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 terminal.go:Flags() - 40ns
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 home.go:12 → /home/kamack38
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → always_enabled: false
[TRACE] 20:52:40.773 terminal.go:GOOS() - 30ns
[DEBUG] 20:52:40.773 segment.go:Execute:109 → segment: Exit
[TRACE] 20:52:40.773 terminal.go:ExecutionTime() - 70ns
[ERROR] 20:52:40.773 terminal.go:FileContent:207 → open /home/kamack38/.aws/config: no such file or directory
[DEBUG] 20:52:40.773 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.773 terminal.go:FileContent(/home/kamack38/.aws/config) - 11.201µs
[TRACE] 20:52:40.773 store.go:](session toggle_cache) - 3.837µs
[DEBUG] 20:52:40.773 properties.go:GetFloat64:52 → threshold: 500.000000
[DEBUG] 20:52:40.773 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.773 terminal.go:StatusCodes() - 60ns
[DEBUG] 20:52:40.773 properties.go:GetString:46 → status_template: {{ .Code }}
[ERROR] 20:52:40.773 terminal.go:CommandPath:308 → exec: "terraform": executable file not found in $PATH
[TRACE] 20:52:40.773 terminal.go:CommandPath(terraform) - 72.005µs
[TRACE] 20:52:40.773 terminal.go:HasCommand(terraform) - 75.391µs
[DEBUG] 20:52:40.773 text.go:patchTemplate:149 → {{ .Code }}
[DEBUG] 20:52:40.773 text.go:patchTemplate:149 →  {{ .Data.Name }} 
[DEBUG] 20:52:40.773 properties.go:GetBool:40 → always_enabled: true
[DEBUG] 20:52:40.773 text.go:patchTemplate:149 →  {{ if gt .Code 0 }}{{ else }}{{ end }} 
[DEBUG] 20:52:40.773 text.go:patchTemplate:149 → {{ if gt .Code 0 }}#FF005D{{ end }}
[DEBUG] 20:52:40.773 text.go:patchTemplate:149 → {{ if gt .Code 0 }}#FF005D{{ end }}
[DEBUG] 20:52:40.775 terminal_unix.go:TerminalWidth:83 → terminal width: 125
[TRACE] 20:52:40.775 terminal_unix.go:TerminalWidth() - 1.129144ms
[DEBUG] 20:52:40.775 engine.go:shouldFill:156 → no filler specified
[TRACE] 20:52:40.775 terminal.go:Shell() - 400ns
[TRACE] 20:52:40.775 terminal.go:Flags() - 130ns
[DEBUG] 20:52:40.775 home.go:12 → /home/kamack38
[TRACE] 20:52:40.775 terminal.go:GOOS() - 50ns
[DEBUG] 20:52:40.775 segment.go:Execute:109 → segment: Text
[DEBUG] 20:52:40.775 store.go:]:150 → (session) found entry: toggle_cache - map[]
[TRACE] 20:52:40.775 store.go:](session toggle_cache) - 9.097µs
[DEBUG] 20:52:40.775 segment.go:isToggled:230 → no toggles found
[TRACE] 20:52:40.775 terminal.go:Shell() - 40ns
[TRACE] 20:52:40.775 terminal.go:Flags() - 30ns
[TRACE] 20:52:40.775 terminal.go:Shell() - 40nsoh-my-posh debug --plain
[TRACE] 20:52:40.775 path.go:Path() - 80ns
```

--- a/src/shell/fish.go
+++ b/src/shell/fish.go
@@ -12,7 +12,7 @@ var fishInit string
 func (f Features) Fish() Code {
 	switch f {
 	case Transient:
-		return "set --global fish_transient_prompt 1"
+		return "set --global _omp_transient_prompt 1"
 	case FTCSMarks:
 		return "set --global _omp_ftcs_marks 1"
 	case PromptMark:

--- a/src/shell/fish_test.go
+++ b/src/shell/fish_test.go
@@ -12,7 +12,7 @@ func TestFishFeatures(t *testing.T) {
 
 	want := `// these are the features
 enable_poshtooltips
-set --global fish_transient_prompt 1
+set --global _omp_transient_prompt 1
 set --global _omp_ftcs_marks 1
 "$_omp_executable" upgrade
 "$_omp_executable" notice

--- a/src/shell/scripts/omp.fish
+++ b/src/shell/scripts/omp.fish
@@ -4,13 +4,19 @@ set --export --global POWERLINE_COMMAND oh-my-posh
 set --export --global CONDA_PROMPT_MODIFIER false
 
 set --global _omp_tooltip_command ''
+set --global _omp_current_rprompt ''
+set --global _omp_transient 0
 set --global _omp_executable ::OMP::
 set --global _omp_ftcs_marks 0
+set --global _omp_transient_prompt 0
 set --global _omp_prompt_mark 0
 
 # disable all known python virtual environment prompts
 set --global VIRTUAL_ENV_DISABLE_PROMPT 1
 set --global PYENV_VIRTUALENV_DISABLE_PROMPT 1
+
+# We use this to avoid unnecessary CLI calls for prompt repaint.
+set --global _omp_new_prompt 1
 
 # template function for context loading
 function set_poshcontext
@@ -38,17 +44,18 @@ end
 function fish_prompt
     set --local omp_status_temp $status
     set --local omp_pipestatus_temp $pipestatus
-
     # clear from cursor to end of screen as
     # commandline --function repaint does not do this
     # see https://github.com/fish-shell/fish-shell/issues/8418
-    # printf \e\[0J
-
-    if contains -- --final-rendering $argv
-        echo -n (_omp_get_prompt transient)
+    printf \e\[0J
+    if test "$_omp_transient" = 1
+        _omp_get_prompt transient
         return
     end
-
+    if test "$_omp_new_prompt" = 0
+        echo -n "$_omp_current_prompt"
+        return
+    end
     set --global _omp_status $omp_status_temp
     set --global _omp_pipestatus $omp_pipestatus_temp
     set --global _omp_no_status false
@@ -88,11 +95,28 @@ function fish_prompt
         iterm2_prompt_mark
     end
 
-    echo -n (_omp_get_prompt primary --cleared=$omp_cleared | string join \n | string collect)
+    # The prompt is saved for possible reuse, typically a repaint after clearing the screen buffer.
+    set --global _omp_current_prompt (_omp_get_prompt primary --cleared=$omp_cleared | string join \n | string collect)
+
+    echo -n "$_omp_current_prompt"
 end
 
 function fish_right_prompt
-    echo -n (_omp_get_prompt right | string join '')
+    if test "$_omp_transient" = 1
+        set --global _omp_transient 0
+        return
+    end
+
+    # Repaint an existing right prompt.
+    if test "$_omp_new_prompt" = 0
+        echo -n "$_omp_current_rprompt"
+        return
+    end
+
+    set --global _omp_new_prompt 0
+    set --global _omp_current_rprompt (_omp_get_prompt right | string join '')
+
+    echo -n "$_omp_current_rprompt"
 end
 
 function _omp_postexec --on-event fish_postexec
@@ -105,6 +129,30 @@ function _omp_preexec --on-event fish_preexec
     if test $_omp_ftcs_marks = 1
         echo -ne "\e]133;C\a"
     end
+end
+
+# perform cleanup so a new initialization in current session works
+if bind \r --user 2>/dev/null | string match -qe _omp_enter_key_handler
+    bind -e \r -M default
+    bind -e \r -M insert
+    bind -e \r -M visual
+end
+
+if bind \n --user 2>/dev/null | string match -qe _omp_enter_key_handler
+    bind -e \n -M default
+    bind -e \n -M insert
+    bind -e \n -M visual
+end
+
+if bind \cc --user 2>/dev/null | string match -qe _omp_ctrl_c_key_handler
+    bind -e \cc -M default
+    bind -e \cc -M insert
+    bind -e \cc -M visual
+end
+
+if bind \x20 --user 2>/dev/null | string match -qe _omp_space_key_handler
+    bind -e \x20 -M default
+    bind -e \x20 -M insert
 end
 
 # tooltip
@@ -136,4 +184,64 @@ end
 function enable_poshtooltips
     bind \x20 _omp_space_key_handler -M default
     bind \x20 _omp_space_key_handler -M insert
+end
+
+# transient prompt
+
+function _omp_enter_key_handler
+    if commandline --paging-mode
+        commandline --function execute
+        return
+    end
+
+    if commandline --is-valid || test -z (commandline --current-buffer | string trim -l | string collect)
+        set --global _omp_new_prompt 1
+        set --global _omp_tooltip_command ''
+
+        if test $_omp_transient_prompt = 1
+            set --global _omp_transient 1
+            commandline --function repaint
+        end
+    end
+
+    commandline --function execute
+end
+
+function _omp_ctrl_c_key_handler
+    if test -z (commandline --current-buffer | string collect)
+        return
+    end
+
+    # Render a transient prompt on Ctrl-C with non-empty command line buffer.
+    set --global _omp_new_prompt 1
+    set --global _omp_tooltip_command ''
+
+    if test $_omp_transient_prompt = 1
+        set --global _omp_transient 1
+        commandline --function repaint
+    end
+
+    commandline --function cancel-commandline
+    commandline --function repaint
+end
+
+bind \r _omp_enter_key_handler -M default
+bind \r _omp_enter_key_handler -M insert
+bind \r _omp_enter_key_handler -M visual
+bind \n _omp_enter_key_handler -M default
+bind \n _omp_enter_key_handler -M insert
+bind \n _omp_enter_key_handler -M visual
+bind \cc _omp_ctrl_c_key_handler -M default
+bind \cc _omp_ctrl_c_key_handler -M insert
+bind \cc _omp_ctrl_c_key_handler -M visual
+
+# legacy functions
+function enable_poshtransientprompt
+    return
+end
+
+# This can be called by user whenever re-rendering is required.
+function omp_repaint_prompt
+    set --global _omp_new_prompt 1
+    commandline --function repaint
 end

